handlers/postitem/postItem_test.go

```diff
@@ -0,0 +1,188 @@
+package postitem
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/mount-joy/thelist-lambda/handlers/testhelpers"
+
+	"github.com/mount-joy/thelist-lambda/data"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPostItemMatch(t *testing.T) {
+	tests := []struct {
+		name        string
+		path        string
+		method      string
+		expectedRes bool
+	}{
+		{
+			name:        "Returns true for a matching path",
+			path:        "/lists/b6cf642d/items/",
+			method:      "POST",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns true for a uppercase ID",
+			path:        "/lists/B6CF642D/items/",
+			method:      "POST",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns true without trailing slash",
+			path:        "/lists/b6cf642d/items",
+			method:      "POST",
+			expectedRes: true,
+		},
+		{
+			name:        "Returns false for list path",
+			path:        "/lists/b6cf642d/",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for list path without trailing slash",
+			path:        "/lists/b6cf642d",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for lists path",
+			path:        "/lists/",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for lists path without trailing slash",
+			path:        "/lists",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for item path",
+			path:        "/lists/b6cf642d-7a72-4969-bcc9-73bb82c4b3f6/items/73bb82c4",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false when path is empty",
+			path:        "",
+			method:      "POST",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a GET request",
+			path:        "/lists/b6cf642d/items/",
+			method:      "GET",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a DELETE request",
+			path:        "/lists/b6cf642d/items/",
+			method:      "DELETE",
+			expectedRes: false,
+		},
+		{
+			name:        "Returns false for a PATCH request",
+			path:        "/lists/b6cf642d/items/",
+			method:      "PATCH",
+			expectedRes: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			dbMocked := &testhelpers.MockDB{}
+			dbMocked.Test(t)
+			defer dbMocked.AssertExpectations(t)
+
+			input := testhelpers.CreateAPIGatewayV2HTTPRequest(tt.path, tt.method, "")
+			d := postItem{db: dbMocked}
```

Comment:
Probably don't need the db at all to be set for these tests

Reply:
I kinda see this as a bit of an implementation detail, i.e. if it wanted to use the DB for this for whatever reason, it could 🤷

Reply:
yeah I have no strongopinion on this either way